mmseg/models/segmentors/seg_tta.py

```diff
@@ -39,11 +39,18 @@ def merge_preds(self, data_samples_list: List[SampleList]) -> SampleList:
                             ).to(logits).squeeze(1)
             else:
                 seg_pred = logits.argmax(dim=0)
-            data_sample = SegDataSample(
-                **{
-                    'pred_sem_seg': PixelData(data=seg_pred),
-                    'gt_sem_seg': data_samples[0].gt_sem_seg
-                })
+            if hasattr(data_samples[0], 'gt_sem_seg'):
+                data_sample = SegDataSample(
+                    **{
+                        'pred_sem_seg': PixelData(data=seg_pred),
+                        'gt_sem_seg': data_samples[0].gt_sem_seg
+                    })
+            else:
+                data_sample = SegDataSample(
+                    **{
+                        'pred_sem_seg': PixelData(data=seg_pred),
+                        'gt_sem_seg': PixelData(data=seg_pred)
```

Comment:
Can we delete this field?

Reply:
I'm afraid not. seg_pred is the final prediction, we will get the last prediction in data_samples if we delete it. But we can still simplify the code by replacing
```suggestion
                       if hasattr(data_samples[0], 'gt_sem_seg'): 
                data_sample = SegDataSample(
                    **{
                        'pred_sem_seg': PixelData(data=seg_pred),
                        'gt_sem_seg': data_samples[0].gt_sem_seg
                    })
            else:
                data_sample = SegDataSample(
                    **{
                        'pred_sem_seg': PixelData(data=seg_pred),
                        'gt_sem_seg': PixelData(data=seg_pred)
                    })
```
 with
```suggestion
            data_sample.set_data({'pred_sem_seg': PixelData(data=seg_pred)})
            if hasattr(data_samples[0], 'gt_sem_seg'):
                data_sample.set_data({'gt_sem_seg': data_samples[0].gt_sem_seg})
```

Reply:
I'm afraid not. seg_pred is the final prediction, we will get the last prediction in data_samples if we delete it. But we can still simplify the code by replacing
```
if hasattr(data_samples[0], 'gt_sem_seg'):
    data_sample = SegDataSample(
        **{
            'pred_sem_seg': PixelData(data=seg_pred),
            'gt_sem_seg': data_samples[0].gt_sem_seg
        })
else:
    data_sample = SegDataSample(
        **{
            'pred_sem_seg': PixelData(data=seg_pred),
            'gt_sem_seg': PixelData(data=seg_pred)
```
with
```
data_sample.set_data({'pred_sem_seg': PixelData(data=seg_pred)})
if hasattr(data_samples[0], 'gt_sem_seg'):
    data_sample.set_data({'gt_sem_seg': data_samples[0].gt_sem_seg})
```

Reply:
Sorry I wasn't clear, I just wanted to delete the field `gt_sem_seg`, because we want to make sure the semantics of these variables are correct.

Reply:
Sorry to misunderstood. Sure, very rigorous idea, we should do this.  And the less equivalent code has also been tested, it may be more concise.